src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -431,12 +753,33 @@ class module_elf : public module_impl
     return true;
   }
 
+  bool
+  patches(uint8_t* base, const std::string& argnm, uint64_t patch) override
+  {
+    auto it = m_arg2patchers.find(argnm);
+    if (it == m_arg2patchers.end())
+      return false;
+
+    it->second.patch(base, patch);
+    return true;
+  }
+
+  const uint8_t&
+  get_os_abi() const override
+  {
+    return m_os_abi;
+  }
+
 public:
   module_elf(xrt::elf elf)
-    : module_impl{elf.get_cfg_uuid()}
+    : module_impl{ elf.get_cfg_uuid() }
     , m_elf(std::move(elf))
-    , m_ctrlcodes{initialize_column_ctrlcode(xrt_core::elf_int::get_elfio(m_elf))}
-    , m_arg2patcher{initialize_arg_patchers(xrt_core::elf_int::get_elfio(m_elf), m_ctrlcodes)}
+    , m_os_abi{ xrt_core::elf_int::get_elfio(m_elf).get_os_abi() }
+    , m_ctrlcodes{ initialize_column_ctrlcode(xrt_core::elf_int::get_elfio(m_elf)) }
+    , m_arg2patcher{ initialize_arg_patchers(xrt_core::elf_int::get_elfio(m_elf), m_ctrlcodes) }
```

Comment:
What is the different between the two m_arg2patcher(s) ?  Do we need both?
E.g. do we need the old original m_ctrlcodes and m_arg2patchers?

Reply:
m_arg2patcher is for Turin
m_arg2patchers is for IPU.
A follow-up PR will merge them after testing on Turin